lib/kafka/consumer.rb

```diff
@@ -57,6 +57,17 @@ def initialize(cluster:, logger:, instrumenter:, group:, offset_manager:, sessio
 
       # The maximum number of bytes to fetch from a single partition, by topic.
       @max_bytes = {}
+
+      # Hash containing offsets for each topic and partition that has the
+      # automatically_mark_as_processed feature disabled. Offset manager is only active
+      # when everything is suppose to happen automatically. Otherwise we need to keep track of the
+      # offset manually in memory for all the time
+      # The key structure for this equals an array with topic and partition [topic, partition]
+      # The value is equal to the offset of the last message we've received
+      # @note It won't be updated in case user marks message as processed, because for the case
+      #   when user commits message other than last in a batch, this would make ruby-kafka refetch
+      #   some already consumed messages
+      @current_offsets = {}
```

Comment:
Can this follow the convention of nested hashes used elsewhere in the code?

Reply:
Yes. Could you provide me with an example just to be sure.

Reply:
```ruby
@current_offsets = Hash.new {|h, k| h[k] = {} }
```

Reply:
@dasch but this type of nested hash presents a different behavior than that I'm aiming and disallows for nil check (as the hash is never nil). I can do that but it will require really inelegant changes for comparing the non-existing content with an empty hash (or something like that)

Reply:
The nil check is done per partition, right?

```ruby
@current_offsets[topic].key?(partition)
```

Reply:
Ah you mean that type of structure. Now I get it. Will update soon